fe/fe-core/src/main/java/org/apache/doris/catalog/Index.java

```diff
@@ -30,36 +31,55 @@
 import java.io.DataOutput;
 import java.io.IOException;
 import java.util.ArrayList;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
 
 /**
  * Internal representation of index, including index type, name, columns and comments.
  * This class will used in olaptable
  */
 public class Index implements Writable {
+    public static final int INDEX_ID_INIT_VALUE = -1;
+
+    @SerializedName(value = "indexId")
+    private int indexId;
```

Comment:
How to do the forward compatibility? that the old index does not has an ID.

Reply:
set default value to -1 for compatibility